packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Snippets/Snippet.Optional.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Providers;
+
+namespace Microsoft.Generator.CSharp.Snippets
+{
+    public static partial class Snippet
+    {
+        public class Optional
+        {
+            public static BoolSnippet IsCollectionDefined(TypedSnippet collection)
+            {
+                return OptionalProvider.Instance.IsCollectionDefined(collection);
```

Comment:
`OptionalProvider` is internal, hence why I added these snippets to be consumed in the SCM plugin.

Reply:
I think we can make optional public